### ISSUE TITLE
Type augmentation not working after v4.12.3

### ISSUE THREAD

**Issue by MishaFrenkman** [nextauthjs/next-auth#5576]
### Question 💬

Hi,

I'm using type augmentation for the Session as described in the docs and everything worked fine. 
After upgrading to version > v4.12.3 it's somehow not getting picked up anymore. 
Is that a bug, or did the api change without the docs being updated?

Best,
Michael

### How to reproduce ☕️

Augment Session

```ts
declare module 'next-auth' {
  /**
   * Returned by `useSession`, `getSession` and received as a prop on the `SessionProvider` React Context
   */
  interface Session {
    user: {
     admin: Boolean
    } & DefaultSession['user'];
  }
}
```

Check in `[...nextauth].ts` if session is augmented inside callback method

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
Should be fixed in `4.14.0`, please give it a try!

**Comment by MishaFrenkman** [comment]
@balazsorban44, still doesn't work for me :(

**Comment by nowlena** [comment]
> @balazsorban44, still doesn't work for me :(

same

**Comment by hecterbonha** [comment]
> @balazsorban44, still doesn't work for me :(

the same happened with me

**Comment by emrysal** [comment]
Hi @balazsorban44 the same thing happens when I try to upgrade next-auth on https://github.com/calcom/cal.com/pull/5622

Between 4.10.3 <> 4.17.0 (narrowed down further by others, 4.10.3 <> 4.12.4) there seems to be a breaking change that disables type augmentation.

**Comment by emrysal** [comment]
So it turns out from 4.13.0 onwards you need to augment the `User` interface. (whereas before only session)

**Comment by grikomsn** [comment]
Tested on TypeScript 4.9.x, possible solution is to augment the adapter and core types like below:

```ts
import type { User as PrismaUser } from "@prisma/client";

declare module "next-auth/adapters" {
  interface AdapterUser extends PrismaUser {}
}

declare module "next-auth/core/types" {
  interface User extends PrismaUser {}
}
```

**Comment by hainguyents13** [comment]
instead of ```declare module "next-auth"```, using ```declare module "next-auth/core/types"``` works for me (thanks @grikomsn)

**Comment by max-sym** [comment]
Having the same issue. @hainguyents13 your solution didn't work for me.

**Comment by north-leshiy** [comment]
> instead of `declare module "next-auth"`, using `declare module "next-auth/core/types"` works for me (thanks @grikomsn)

In version 4.24.7 you must also use `src`

```ts
declare module 'next-auth/src/core/types' {
  interface Session extends DefaultSession {
    // ...
  }
}
```

**Comment by udiskie** [comment]
hello everyone. I have expeperienced the same issue, and finally i solved as it follows:
**src/app/api/auth/[...nextauth]/route.ts:**
`declare module "next-auth" {
  interface Session extends DefaultSession {
    user: DefaultSession["user"] & {
      data: {
        id: number,
        isActive: boolean,
        createdAt: string,
        updatedAt: string,
        name: string,
        password: string,
        email: string,
        role: string,
        isSubscribed: boolean,
        lang: string,
        passwordKey: any,
        company: {
          id: number,
          isActive: boolean,
          createdAt: string,
          updatedAt: string,
          name: string,
          createdBy: number
        },
      }
    }
  }
}`

Basically, I just added that  into my next auth api route handler, and all missed prop issues regarding session disappeared. For my particular case i needed to add that "data" prop that holds that object.

I am a beginner regarding next and next auth, so i am not really sure why this works, if someone can explain it to me it would be awesome
